tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -0,0 +1,757 @@
+
+
+class InfiniteRetentionTest(PreallocNodesTest):
+    DEFAULT_PARAMS = InfiniteRetentionParameters()
+    FINJECT_PARAMS = InfiniteRetentionParameters(segment_upload_interval=45,
+                                                 manifest_upload_interval=20,
+                                                 grace_upload_wait_interval=60,
+                                                 producer_ack_timeout=180)
+
+    def __init__(self, test_context, *args, **kwargs):
+        self.params = self.DEFAULT_PARAMS
+
+        self.si_settings = SISettings(
+            test_context=test_context,
+            log_segment_size=self.params.segment_size,
+        )
+        kwargs['si_settings'] = self.si_settings
+
+        # Use interval uploads so that at end of test we may do an "everything
+        # was uploaded" success condition.
+        kwargs['extra_rp_conf'] = {
+            # We do not intend to do interval-triggered uploads during produce,
+            # but we set this property so that at the end of the test we may
+            # do a simple "everything was uploaded" check after the interval.
+            'cloud_storage_segment_max_upload_interval_sec':
+            self.params.segment_upload_interval,
+            # The test will assert that the number of manifest uploads does
+            # not exceed what we would expect based on this interval.
+            'cloud_storage_manifest_max_upload_interval_sec':
+            self.params.manifest_upload_interval,
+            # Default retention is infinite
+            'delete_retention_ms':
+            self.params.retention_ms,
+            # enable merging to generate more re-uploads
+            'cloud_storage_enable_segment_merging':
+            self.params.cloud_storage_enable_segment_merging,
+
+            # segment merging params
+            'cloud_storage_segment_size_target':
+            self.params.cloud_storage_segment_size_target,
+            'cloud_storage_segment_size_min':
+            self.params.cloud_storage_segment_size_min,
+
+            # clean out local data more frequently
+            'cloud_storage_housekeeping_interval_ms':
+            self.params.cloud_storage_housekeeping_interval_ms,
+
+            # rps should not hit the limit set here
+            'cloud_storage_idle_threshold_rps':
+            self.params.cloud_storage_idle_threshold_rps,
+
+            # manifest cache reduced
+            'cloud_storage_manifest_cache_size':
+            self.params.cloud_storage_manifest_cache_size,
+
+            # evict it frequently
+            'cloud_storage_materialized_manifest_ttl_ms':
+            self.params.cloud_storage_materialized_manifest_ttl_ms,
+
+            # This causes spillover to hapen more often
+            # 'cloud_storage_spillover_manifest_max_segments':
+            # self.params.cloud_storage_spillover_manifest_max_segments,
+        }
+        super().__init__(test_context, node_prealloc_count=1, *args, **kwargs)
```

Comment:
I was reminded, looking at redpanda.py, that you may find it useful for this test to run with `disable_cloud_storage_diagnostics=True`

Reply:
What different would it make in our case?